datasets/me/acp_peps/crawler.py

```diff
@@ -0,0 +1,162 @@
+import io
+from csv import DictReader
+from datetime import datetime
+from typing import List, Dict, Optional, Tuple
+from zipfile import ZipFile, BadZipFile
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+from zavod.shed.trans import (
+    apply_translit_full_name,
+    make_position_translation_prompt,
+)
+
+TRANSLIT_OUTPUT = {
+    "eng": ("Latin", "English"),
+}
+POSITION_PROMPT = prompt = make_position_translation_prompt("cnr")
+
+
+def extract_latest_filing(
+    dates: List[Dict[str, object]]
+) -> Tuple[Optional[str], Optional[int]]:
+    if not dates:
+        return None, None
+
+    # Sort the dates in descending order
+    dates_sorted = sorted(
+        dates,
+        key=lambda item: datetime.strptime(item["datum"], "%Y-%m-%d"),
+        reverse=True,
+    )
+
+    # Assume the first entry is the latest
+    latest_entry = dates_sorted[0]
+    latest_date: Optional[str] = latest_entry.get("datum")
+    latest_report_id: Optional[int] = latest_entry.get("stariIzvjestaj")
```

Comment:
I think this can be handled by the loop, cant it?

```
for entry in dates_sorted:
    if report_id := entry.get("starIzcjestaj") != "-1":
        return entry.get(datum), report_id
return None, None # or just return the entry?